### PR TITLE
Add class documentation.

### PR DESCRIPTION
**Pull request by chrrrisw**
Hi,
I have added documentation via Sphinx autodoc, and attempted to document the main classes.

Let me know what you think.

Regards,
Chris.

### CODE REVIEW

File: hsaudiotag/id3v2.py [hunk 1]
```diff
@@ -264,34 +268,39 @@ def _get_frame_text_line(self, frame_id):
         return result.replace('\n', ' ').replace('\r', ' ')
 
     # --- Properties
-    size = property(lambda self: self._header.tagsize)
-    data_size = property(lambda self: self._header.datasize)
-    exists = property(lambda self: self._header.valid)
-    flags = property(lambda self: self._header.hflags)
-    version = property(lambda self: self._header.vmajor)
+    size = property(lambda self: self._header.tagsize)  #: The size of the file, in bytes.
+    data_size = property(lambda self: self._header.datasize)  #: Foo
```
Comment (hsoft) [inline hsaudiotag/id3v2.py]: Why "Foo"?

File: hsaudiotag/auto.py [hunk 2]
```diff
@@ -67,6 +89,7 @@ def _set_attrs(self, f):
                 setattr(self, attrname, getattr(tag, attrname))
 
     def _set_invalid_attrs(self):
+        """"""
```
Comment (hsoft) [inline hsaudiotag/auto.py]: Why empty?

**Comment by hsoft** [comment]
Thanks! That's some serious documentation! I personally wouldn't have gone this verbosly, but now that it's done, let's not throw it away.

When I build the documentation, I get tons of `more than one target found for cross-reference` errors. Do you also get them?

**Comment by chrrrisw** [comment]
Sorry about the Foo's - I'll put something more meaningful there - and remove empty documentation on private methods.

**Comment by chrrrisw** [comment]
My run gives me no errors - perhaps the build directory is clashing with the hsaudiotag directory.

I run the documentation build thusly:
python3 setup.py build_sphinx

**Comment by hsoft** [comment]
really? With a clean git repo and sphinx 1.3.1 installed globally, `python3 setup.py build_sphinx` gives me a bunch of warnings looking like:

    /home/vdupras/src/hsaudiotag/hsaudiotag/wma.py:docstring of hsaudiotag.wma.WMADecoder:None: WARNING: more than one target found for cross-reference 'year': hsaudiotag.id3v2.Id3v2.year, hsaudiotag.mp4.File.year

I don't think it's a directory clash since the clash seems to be between different classes with similar `ivar`s. You really don't have that?

**Comment by chrrrisw** [comment]
I've got sphinx 1.2.2 installed and don't get that - I'll update and see what happens.

Apologies for the verbosity - it comes from years of programming for the military.

**Comment by chrrrisw** [comment]
OK, I'm getting the errors with version 1.4.1 - will investigate.

**Comment by chrrrisw** [comment]
Hopefully all should be good now.

Sphinx was trying to cross-reference instance variable names - I've stopped it using more fully qualified names.

**Comment by hsoft** [comment]
Thanks!

**Pull request merged**